kong/db/dao/plugins.lua

```diff
@@ -89,18 +89,23 @@ end
 
 
 function Plugins:update(primary_key, entity, options)
-  options = options or {}
-  options.hide_shorthands = true
-  local rbw_entity = self.super.select(self, primary_key, options) -- ignore errors
-  if rbw_entity then
-    entity = self.schema:merge_values(entity, rbw_entity)
-  end
-  local ok, err, err_t = check_protocols_match(self, entity)
-  if not ok then
-    return nil, err, err_t
+  if entity.protocols or entity.service or entity.route then
+    if (entity.protocols and not entity.route)
+    or (entity.service and not entity.protocols)
+    or (entity.route and not entity.protocols)
```

Comment:
What's the idea behind this logic? 🤔 Why protocols and routes are XORed etc.?

Reply:
same question: shouldn't we check for "protocol match" even when both e.g. protocols and route were updated?

Reply:
@nowNick, @samugi, yes we are checking, but in that case we are skipping RBW.

Reply:
you're right I misinterpreted the scope of that if block, my bad.